src/almanack/metrics/data.py

```diff
@@ -1182,3 +1195,71 @@ def find_doi_citation_data(repo: pygit2.Repository) -> Dict[str, Any]:
                 LOGGER.warning(f"Error during OpenAlex exact DOI lookup: {e}")
 
     return result
+
+
+def compute_sustainability_score(
+    table: List[Dict[str, Union[int, float, bool]]]
+) -> float:
+    """
+    Computes a sustainability score by normalizing
+    numeric metrics and incorporating boolean metrics.
+    Adjusts for metrics that should be inversely
+    proportional to the sustainability score.
+
+    Args:
+        data (List[Dict[str, Union[int, float, bool]]]):
+            A list of dictionaries containing metrics.
+            Each dictionary must have a "result" key
+            with a value that is an int, float, or bool.
+              Optionally, a "direction" key may be
+            included for numeric values to specify the
+            relationship to sustainability:
+            - 1 (positive correlation)
+            - 0 (no correlation)
+            - -1 (negative correlation)
+
+    Returns:
+        float:
+            The computed sustainability score, normalized between 0 and 1.
+    """
+    results = []
+    bool_results = []
+
+    # Gather numeric and boolean values with a direction
+    for item in table:
+        if isinstance(item["result"], bool) and item["direction"] != 0:
+            # for direction == 1 we treat True as positive correction
+            # and False as a negative correlation.
+            if item["direction"] == 1:
+                bool_results.append(1 if item["result"] else 0)
+            # for direction == -1 we treat True as negative correction
+            # and False as a positive correlation.
+            elif item["direction"] == -1:
+                bool_results.append(0 if item["result"] else 1)
+
+        elif isinstance(item["result"], (int, float)) and item["direction"] != 0:
+            # Numeric values inverted or kept as-is based on direction
+            results.append(
+                item["result"] if item["direction"] == 1 else 1 - item["result"]
+            )
+
+    # Normalize numeric values if any
+    if results:
+        scaler = RobustScaler()
```

Comment:
why did you choose this scaler?

Reply:
Thanks for checking on this - I began work with a [MinMaxScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html) and found that it was producing values that were out of alignment with one another due to outliers and how the metrics scale individually. For example, we might have `{metric1: 1, metric2: 1000000, metric3: -1000}`. I explored [RobustScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.RobustScaler.html) as a result of this which states upfront: "Scale features using statistics that are robust to outliers." and found that I got better results. I should test this assumption better and do a further analysis here. I'll follow up with more thoughts based on what I find. Perhaps there is existing work which we can cite as part of this.

Reply:
RobustScaler has now been removed along with the removal of numeric metrics within this function. [See here for more](https://github.com/software-gardening/almanack/pull/208#discussion_r1896969340).